### PR TITLE
rsx: Minor fixup for image_in

### PR DESCRIPTION
**Pull request by kd-11**
- Fix typo (scale_x->scale_y)
- Remove convoluted temp buffer creation and just use vector allocator instead

### CODE REVIEW

File: rpcs3/Emu/RSX/rsx_methods.cpp [hunk 1]
```diff
@@ -1018,17 +1018,17 @@ namespace rsx
 					return;
 			}
 
-			std::unique_ptr<u8[]> temp1, temp2, temp3, sw_temp;
+			std::vector<u8> temp1, temp2, temp3, sw_temp;
 
 			if (scale_y < 0 || scale_x < 0)
 			{
-				temp1.reset(new u8[in_pitch * (in_h - 1) + (in_bpp * in_w)]);
+				temp1.resize(in_pitch * in_h);
```
Comment (elad335) [inline rpcs3/Emu/RSX/rsx_methods.cpp]: What if in_pitch < in_bpp * in_w? can't it result in requested size of allocation lower than needed?

Comment (kd-11) [reply rpcs3/Emu/RSX/rsx_methods.cpp]: That is an impossible definition of a 2D region. I can add an assert for it though.

Comment (elad335) [reply rpcs3/Emu/RSX/rsx_methods.cpp]: Maybe its not useful for games but im pretty sure it is allowed.

Comment (elad335) [reply rpcs3/Emu/RSX/rsx_methods.cpp]: I tested it, its allowed.

Comment (elad335) [reply rpcs3/Emu/RSX/rsx_methods.cpp]: For clarification its only about in_pitch, with out_pitch its causing an exception to be raised.

Comment (kd-11) [reply rpcs3/Emu/RSX/rsx_methods.cpp]: This is old code. I rewrote it in a better way. There is still another somewhat unrelated bug but I have commits for them locally.

**Pull request merged**